### ISSUE TITLE
Blank screen to do with MySqli constants

### ISSUE THREAD

**Issue by RegProctor** [phpmyadmin/phpmyadmin#17867]
### Describe the bug

I just upgraded my openSUSE Leap from 15.3 to 15.4. In the process it upgraded phpMyAdmin to 5.2 and then all as I could get is a blank screen. The system is running PHP 7.4.

I tried PMA 5.1.4, down to 5.1.1 and got the same result so I settled on 5.1.4 to do some debugging figuring it was likely the most solid.

After much figuring out I got it to work but I had to do the following:

In DbiMysqli.php:

Take the constants in the array:
```php
    private static $flagNames = [

    MYSQLI_NUM_FLAG             => 'num',
    MYSQLI_PART_KEY_FLAG        => 'part_key',
    MYSQLI_SET_FLAG             => 'set',
    MYSQLI_TIMESTAMP_FLAG       => 'timestamp',
    MYSQLI_AUTO_INCREMENT_FLAG  => 'auto_increment',
    MYSQLI_ENUM_FLAG            => 'enum',
    MYSQLI_ZEROFILL_FLAG        => 'zerofill',
    MYSQLI_UNSIGNED_FLAG        => 'unsigned',
    MYSQLI_BLOB_FLAG            => 'blob',
    MYSQLI_MULTIPLE_KEY_FLAG    => 'multiple_key',
    MYSQLI_UNIQUE_KEY_FLAG      => 'unique_key',
    MYSQLI_PRI_KEY_FLAG         => 'primary_key',
    MYSQLI_NOT_NULL_FLAG        => 'not_null',
];
```

And move then to the root index.php file so they were not under a namespace like:

```php
global $route, $containerBuilder;

$dbiMysqli_flagNames = [
    MYSQLI_NUM_FLAG             => 'num',
    MYSQLI_PART_KEY_FLAG        => 'part_key',
    MYSQLI_SET_FLAG             => 'set',
    MYSQLI_TIMESTAMP_FLAG       => 'timestamp',
    MYSQLI_AUTO_INCREMENT_FLAG  => 'auto_increment',
    MYSQLI_ENUM_FLAG            => 'enum',
    MYSQLI_ZEROFILL_FLAG        => 'zerofill',
    MYSQLI_UNSIGNED_FLAG        => 'unsigned',
    MYSQLI_BLOB_FLAG            => 'blob',
    MYSQLI_MULTIPLE_KEY_FLAG    => 'multiple_key',
    MYSQLI_UNIQUE_KEY_FLAG      => 'unique_key',
    MYSQLI_PRI_KEY_FLAG         => 'primary_key',
    MYSQLI_NOT_NULL_FLAG        => 'not_null',
];
```

And then reference the global variable:
```php
    private static $flagNames = [];

    public function __construct() {
      global $dbiMysqli_flagNames;
      self::$flagNames = $dbiMysqli_flagNames;
    }
```

Before I did the above I was able to get past the blank screen and just have warnings spit out by commenting out the "use const" declarations like so:
//use const MYSQLI_ENUM_FLAG;
//use const MYSQLI_MULTIPLE_KEY_FLAG;
//use const MYSQLI_NOT_NULL_FLAG;
...

Once I moved the reference to the constants outside the namespace the constants declaration could be uncommented without the output going blank.

I find the need for this fix strange as the way it was seems to me like it should work to me so perhaps there is something quirky with the Leap's 15.4 PHP 7.4 build.

I also had to bypass the check for mysqli_connect because the function_exists() function wasn't working in the class but returned true just fine if tested outside the class.

### To Reproduce

Not sure if this is reproducible on another system but I gave a fix above so hopefully that is enough... although I wouldn't recommend my fix as "the fix", it was quick and dirty, but only as a way to know what's causing the problem.

### Expected behavior
To work.

### Server configuration

- Operating system: OpenSUSE Leap 15.4
- Web server: apache 2.4
- Database version: mariaDB 10
- PHP version: 7.4
- phpMyAdmin version: 5.1.1 up to 5.2

### Client configuration

- Browser: FF and Chrome
- Operating system: Leap 15.4

### Additional context

It's all local on the same system, nothing remote going on.

The URL is www.pma.local.

**Comment by williamdes** [comment]
Hi @RegProctor 
What is the output of `php -m` ?
It looks like you have a broken build of PHP with mysqli

**Comment by RegProctor** [comment]
Okay, it's starting to make sense and this might be an apache/PHP config issue because this:

<?php
echo "\n\n";
if(defined('MYSQLI_NUM_FLAG')) {
echo 'Exists';} 
else {echo 'Absent';}
echo "\n\n";

Outputs "Exists" from the command line as in:
>php test-stuff.php 

but "Absent" from the associated URL as in:   http://www.pma.local/test-stuff.php .

I came across this because I started getting PHP Warning: Use of undefined constant errors for the constants I moved such as: 

MYSQLI_NUM_FLAG

which I wasn't seeing before. Now I just have to figure out what apache/PHP configuration might be messed up to cause this.

**Comment by RegProctor** [comment]
> Hi @RegProctor What is the output of `php -m` ? It looks like you have a broken build of PHP with mysqli

Yeah, I checked for mysqli being there many times. Here's the output
[pi.txt](https://github.com/phpmyadmin/phpmyadmin/files/9937066/pi.txt)

**Comment by RegProctor** [comment]
> It looks like you have a broken build of PHP with mysqli

As a logical next thing to check here's grep for mysqli.so on the php configuration:

```
quark:/etc/php7 # grep mysqli.so -rn *
apache2/php.ini.rpmsave:879:;   extension=/path/to/extension/mysqli.so
apache2/php.ini:901:;   extension=/path/to/extension/mysqli.so
apache2/php.ini.rpmnew:900:;   extension=/path/to/extension/mysqli.so
cli/php.ini.rpmsave:879:;   extension=/path/to/extension/mysqli.so
cli/php.ini:891:;   extension=/path/to/extension/mysqli.so
cli/php.ini.rpmnew:900:;   extension=/path/to/extension/mysqli.so
conf.d/mysqli.ini:2:extension=mysqli.so
fpm/php.ini:900:;   extension=/path/to/extension/mysqli.so

```

I also ran:

```
<?php
phpinfo();
```


from : http://www.pma.local/info.php

and the mysqli module shows up which I don't understand because I thought if it was absent then I wouldn't see it there I've attached the output from that.
[phpinfo-output.zip](https://github.com/phpmyadmin/phpmyadmin/files/9937154/phpinfo-output.zip)

**Comment by williamdes** [comment]
If it shows absent and present elsewere 🤔 
What is your php fpm and nginx config them ? 🤔

**Comment by RegProctor** [comment]
> If it shows absent and present elsewhere thinking What is your php fpm and nginx config them ? thinking

I've never configured those and don't use them. I just use apache.

**Comment by williamdes** [comment]
> > If it shows absent and present elsewhere thinking What is your php fpm and nginx config them ? thinking
> 
> I've never configured those and don't use them. I just use apache. If these are things I should check just let me know how.

Okay. To summarize:
5.1.4 works, previous versions do not
and 5.2.0 does not work.

If you edit phpMyAdmin's `index.php` and add
```php
<?php
echo "\n\n";
if(defined('MYSQLI_MULTIPLE_KEY_FLAG')) {
echo 'Exists';} 
else {echo 'Absent';}
echo "\n\n";
```

What does it say ?

**Comment by RegProctor** [comment]
> If it shows absent and present elsewhere thinking What is your php fpm and nginx config them ? thinking

I found one more thing that I didn't know about. I was looking through the apache config and there are two files I didn't know about: phpMyAdmin.conf and phpMyAdmin.inc.

Having a quick look at them there are file paths of /usr/share/phpMyAdmin and it's not running there right now because for debugging I wasn't going to mess with something under /usr/share/ so it's in a subfolder of my home directory for the purposes of debugging. I'll change those paths in the apache config phpMyAdmin.* files to where I have it now.

I don't think that will solve it because it wasn't working in the first place before I moved it but at least it should hopefully be one more variable that won't be an issue.

[apache-phpMyAdmin.zip](https://github.com/phpmyadmin/phpmyadmin/files/9937277/apache-phpMyAdmin.zip)

**Comment by RegProctor** [comment]
> What does it say ?

Absent

**Comment by williamdes** [comment]
Hi @kamil-tekiela 
Can I have some of your help on this issue please ?

**Comment by ibennetch** [comment]
I don’t have much to add to this yea, but the output of php -m and
phpinfo() through the web server are not expected to be the same. Using
php-m on the command line (or the output of phpinfo() viewed through the
command line) use the command line configuration, which has a separate
configuration from the PHP Apache module.

So trust what phpinfo() says when viewed through the web server, not the
command line output.

**Comment by RegProctor** [comment]
> > What does it say ?
> 
> Absent

Yeah, I know how you feel, it's been a problem for me for two days now. It's weird.

I changed the file paths in the apache phpMyAdmin.conf config file so they point to where I'm debugging, no change unfortunately.

I was thinking of installing the previous version of PHP 7.3 thinking many suse messed something up with the latest version but I couldn't find any version other than PHP 7.4 for Leap 15.4.

It's late where I am, 3:41 in the morning so I'm going to have to sign off for a while. I'll be back as soon as I start my next day.

**Comment by RegProctor** [comment]
> So trust what phpinfo() says when viewed through the web server, not the command line output.

Correct me if I'm wrong but phpinfo() (attached in a previous comment) is showing that mysqli is there through the web server but when we test that with code in the index.php file it suggests that it is not. Is that correct or am I reading the phpinfo() output incorrectly?

**Comment by RegProctor** [comment]
One last thing, just to make things wierder, I was thinking maybe something in the lamp stack was broken so no website could run so I tried an old Drupal (CMS) website I was working on and the whole thing worked just fine and Drupal uses mysql quite extensively so the lamp stack works in general.

**Comment by kamil-tekiela** [comment]
This is definitely not phpMyAdmin issue. To be blunt, the fix you suggested makes zero sense. Is there an error that got you to this conclusion? What debugging did you do? From your further messages, it seems that you are missing mysqli extension or it's not enabled. Check your error logs and make sure that startup errors are enabled too so that you can see what's wrong with loaded extensions. In the end, you can try to clean reinstall PHP and reenable mysqli extension. From the `phpinfo()` output it seems it's not enabled or it cannot be loaded.

**Comment by RegProctor** [comment]
>  From the `phpinfo()` output it seems it's not enabled or it cannot be loaded.

I saw this in the file:

MySQLi | Zak Greant, Georg Richter, Andrey Hristov, Ulf Wendel

but looking at closer it's under "Module Authors" which just because it's there probably doesn't mean it's loaded.

Agreed, what I did isn't the solution, there's a bug in PHP in my install that's preventing the module from loading when being used in web mode but not a problem with the CLI mode. As for how I got to where I was, if all as you are getting is a blank screen and no errors just getting "something" to latch onto was where I was at and I have never had to look at PMA code before so I was flying blind for a while.

I think my next step is to file a bug report with suse as there is nothing in the config to make it load modules differently between the cli and web mode. Or, I could just load PHP independently of the distro. packages but I really don't want to do that as in the long run that tends to cause problems.

I found my IDE has a data explorer for databases so there's no urgency at this point so waiting for the suse team to figure it out is going to be the best solution.
